internal/provider_test/integration_test.go

```diff
@@ -977,39 +977,50 @@ func TestImmutableAttributeChange(t *testing.T) {
 		HasAttribute("status.ready")
 
-	// Change DBA password and verify that a warning is displayed by
-	// Terraform. Since the DBA password is only sent on create, there will
-	// not be an error from the server. TODO(asz6): When password rotation
-	// is supported by the REST service, we will accept updates to the DBA
-	// password and remove the RequiresReplace plan modifier from the
-	// `dba_password` attribute.
-	t.Run("applyDbaPasswordChange", func(t *testing.T) {
-		updatedPassword := "updated"
-		vars.database.DbaPassword = &updatedPassword
-		tf.WriteConfigT(t, vars.builder.Build())
-		out, err := tf.Apply()
-		require.NoError(t, err)
-		require.Contains(t, string(out), "Immutable Attribute Change")
-		require.Contains(t, string(out), "`dba_password`")
-		require.Contains(t, string(out), "(`<redacted>`)")
-	})
+	// Check whether the REST server supports updating the DBA password
+	client, err := vars.providerCfg.CreateClient()
+	require.NoError(t, err)
+	resp, err := client.UpdateDbaPassword(
+		ctx, vars.database.Organization, vars.database.Project, vars.database.Name, nil,
+		openapi.UpdateDbaPasswordModel{Current: *vars.database.DbaPassword})
+	require.NoError(t, err)
 
-	// Change the DBA password again, this time with
-	// NUODB_CP_ALLOW_DESTRUCTIVE_REPLACE=true
-	t.Run("applyDbaPasswordChangeWithReplace", func(t *testing.T) {
-		t.Setenv(framework.ALLOW_DESTRUCTIVE_REPLACE_VAR, "true")
+	// "404 Not Found" with no "detail" message indicates that /dbaPassword
+	// sub-resource does not exist. Run appropriate test case based on what
+	// behavior is supported.
+	err = helper.ParseResponse(resp, nil)
+	if IsDbaPasswordUpdateUnsupportedError(resp, err) {
+		t.Run("dbaPasswordChangeRejected", func(t *testing.T) {
+			// Change DBA password and run `terraform apply`. Revert
+			// DBA password change when finished.
+			originalPassword := vars.database.DbaPassword
+			vars.database.DbaPassword = ptr("updated")
+			defer func() {
+				vars.database.DbaPassword = originalPassword
+			}()
 
-		updatedPassword := "updated-again"
-		vars.database.DbaPassword = &updatedPassword
-		tf.WriteConfigT(t, vars.builder.Build())
-		out, err := tf.Apply()
+			tf.WriteConfigT(t, vars.builder.Build())
+			out, err := tf.Apply()
+			// Password change should be rejected if the REST server
+			// does not support it
+			require.Error(t, err)
+			require.Contains(t, string(out), "Configured DBA password was changed")
+		})
+	} else {
 		require.NoError(t, err)
-		require.Contains(t, string(out), "# forces replacement")
-		require.Contains(t, string(out), "Apply complete! Resources: 1 added, 0 changed, 1 destroyed.")
-	})
+		t.Run("dbaPasswordChange", func(t *testing.T) {
+			// Change DBA password and run `terraform apply`
+			vars.database.DbaPassword = ptr("updated")
+			tf.WriteConfigT(t, vars.builder.Build())
+			out, err := tf.Apply()
+			require.NoError(t, err)
+			require.Contains(t, string(out), "0 to add, 1 to change, 0 to destroy.")
```

Comment:
Is it possible to also test if the password is actually changed using the REST client via the new `/dbaPassword` sub-resource?

Reply:
Good idea.

Reply:
This actually isn't possible because (1) the integration test does do password update and (2) the end-to-end tests are run without readiness checks for this test case. The update that I just pushed does exercise that the password change was issued correctly via REST, which is all we really need to exercise in Terraform since its only interaction is with the REST service.

Reply:
Okay. Thanks for the explanation!